framework/src/transfers/MultiAppNearestNodeTransfer.C

```diff
@@ -411,7 +413,7 @@ MultiAppNearestNodeTransfer::execute()
       {
         MooseVariableFEBase & from_var = _from_problems[_cached_froms[i_proc][qp]]->getVariable(
             0,
-            _from_var_name,
+            _from_var_name[0],
```

Comment:
This is quite annoying. We can create `_from_var_name` in the base class and rename the vector one as `_from_var_names`. The `_from_var_name` is equal to `_from_var_names[0]`. Another suggestion is to create a call back function in the base class like `bool allowMultipleVariables() {return true; }` and let those transfers do not support multiple variables `return false`.

Reply:
Good point.